frontends/common/resolveReferences/resolveReferences.h

```diff
@@ -23,12 +23,20 @@ limitations under the License.
 #include "lib/cstring.h"
 
 namespace P4 {
+
+/// Helper class to indicate types of nodes that may be returned during resolution.
 enum class ResolutionType {
     Any,
     Type,
     TypeVariable
 };
 
+/**
+ *  Data structure used while resolving references that comprises:
+ *  - A stack of namespaces
```

Comment:
I think that the description of the fields belongs to the fields, not to the class.
This class is logically a stack of nested namespaces where the resolution of a symbol is attempted, inside-out.